app/routes.test.js

```diff
@@ -0,0 +1,402 @@
+import request from 'supertest';
+import { App } from '../app';
+import routes from './routes';
+import * as dashboardControllers from './pages/dashboard/controller';
+import * as sectionControllers from './pages/section/controller';
+import * as subsectionControllers from './pages/dashboard/subDashboards/controller';
+import * as previewControllers from './pages/preview/controller';
+
+jest.mock('./logger');
+
+const mockDashboardContext = {
+  title: 'Dashboard title',
+  mainAdvice: undefined,
+  additionalAdvice: undefined,
+  previewUrl: '/',
+  submitForModerationUrl: '/',
+  returnToDashboardUrl: '/',
+  errors: undefined,
+  sectionGroups: [
+    {
+      id: 'sectiongroup-id',
+      title: 'Sectiongroup Title',
+      sections: [],
+    },
+  ],
+};
+const mockSectionContext = {
+  title: 'Section Title',
+  submitActionUrl: '/s',
+  mainAdvice: 'Section Main Advice',
+  additionalAdvice: [
+    'Section additional advice',
+  ],
+  warningAdvice: undefined,
+  returnToDashboardUrl: '/',
+  submitText: 'submitText',
+  errors: undefined,
+  questions: [
+    {
+      id: 'a-question',
+      type: 'bulletpoint-list',
+      mainAdvice: undefined,
+      additionalAdvice: undefined,
+      footerAdvice: undefined,
+      fields: [],
+    },
+  ],
+};
+const mockSectionPostData = {
+  'a-question': [
+    'bullet 1',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+  ],
+};
+const mockPostSectionContext = {
+  title: 'Post Section Title',
+  submitActionUrl: '/',
+  mainAdvice: 'post section main advice',
+  additionalAdvice: [
+    'post section additional advice',
+  ],
+  warningAdvice: undefined,
+  returnToDashboardUrl: '/',
+  submitText: 'submitText',
+  errors: [
+    {
+      text: 'This is over the character limit',
+      href: '#',
+    },
+  ],
+  questions: [
+    {
+      id: 'a-question',
+      type: 'bulletpoint-list',
+      mainAdvice: undefined,
+      additionalAdvice: undefined,
+      footerAdvice: undefined,
+      fields: [],
+    },
+  ],
+};
+const mockPreviewContext = {
+  sections: {
+    'solution-description': { answers: [Object] },
```

Comment:
Can we do `'solution-description': { answers: {} },` ?

Reply:
yep - I did this for the rest but must have missed this one :)